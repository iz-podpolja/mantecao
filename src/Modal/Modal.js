import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import Popover from '../Popover'
import { colors } from '../variables'

class Modal extends Component {
  render() {
    const { children, show, onClose } = this.props
    return (
      <Popover show={show} showBackdrop onClose={onClose}>
        {children}
      </Popover>
    )
  }
}

Modal.propTypes = {
  show: PropTypes.bool,
  onClose: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.object])
}

export default Modal
