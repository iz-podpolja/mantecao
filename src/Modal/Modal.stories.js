import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import Modal from './Modal'

storiesOf('Modal', module).add('default', () =>
  <Modal show onClose={action('onClose')}>
    <div style={{ background: '#FFF', padding: '1em' }}>
      default modal
    </div>
  </Modal>
)
