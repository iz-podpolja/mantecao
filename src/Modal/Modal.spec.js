import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import Modal from './Modal'

describe('Modal Component', () => {
  it('renders properly', () => {
    const tree = renderer
      .create(<Modal onClose={() => false}><div>child</div></Modal>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
