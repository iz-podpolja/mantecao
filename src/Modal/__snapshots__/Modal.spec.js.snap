// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component renders properly 1`] = `
<div
  className="sc-bxivhb tHErp"
  onClick={[Function]}
>
  <div
    className="sc-bdVaJa bKBZyW"
  >
    <div
      className="sc-bwzfXH dJAmok"
      onClick={[Function]}
    >
      <div>
        child
      </div>
    </div>
  </div>
</div>
`;
